packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const error = require('../error');
+const isA = require('joi');
+const ScopeSet = require('fxa-shared').oauth.scopes;
+const validators = require('./validators');
+
+module.exports = (log, db, config, customs, push, oauthdb, subhub) => {
+  const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
+    'https://identity.mozilla.com/apps/subscriptions';
+
+  async function handleAuth(auth) {
+    const scope = ScopeSet.fromArray(auth.credentials.scope);
+    if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+      throw error.invalidToken('Invalid authentication scope in token');
+    }
+    const uid = auth.credentials.user;
+    const account = await db.account(uid);
+    const email = account.primaryEmail.email;
+    return { scope, uid, email };
+  }
+
+  return [
+    {
+      method: 'GET',
+      path: '/oauth/subscriptions/plans',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        response: {
+          schema: isA.object().keys({
+            plan_id: validators.subscriptionsPlanId.required(),
+            product_id: validators.subscriptionsProductId.required(),
+            interval: isA.string().required(),
+            amount: isA.number().required(),
+            currency: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listPlans', request);
+        await handleAuth(request.auth);
```

Comment:
I wonder if there is a way to make `handleAuth` an [auth strategy](https://github.com/lmorchard/fxa/blob/96a000ea6e702e7474e7b016e692b19cc7db9d0d/packages/fxa-auth-server/lib/server.js#L341) so it doesn't need to be called in each of these routes. I'm not sure whether the required scopes can be passed as data to the auth-strategy somehow.